### ISSUE TITLE
How to monitor the training process

### ISSUE THREAD

**Issue by Guocode** [NVlabs/SSV#4]
After fix some bugs, I've successfully begin to train the network, and how can I determine whether the result is ok from the log like this 
```
  8%|███▎                                     | 84/1034 [04:06<46:25,  2.93s/it]G: -0.9; G_z: 1.3; G_vp: 8.8; D: -0.3; D_z: 1.3; D_vp: 9.1; CosD: -0.166; FL_c: 9.332; Grad: 3.9
  8%|███▎                                     | 85/1034 [04:09<46:20,  2.93s/it]G: -0.8; G_z: 1.3; G_vp: 9.5; D: -0.5; D_z: 1.2; D_vp: 9.7; CosD: -0.163; FL_c: 8.553; Grad: 4.4
  8%|███▍                                     | 86/1034 [04:11<46:16,  2.93s/it]G: -1.1; G_z: 1.2; G_vp: 9.1; D: -0.6; D_z: 1.3; D_vp: 9.3; CosD: -0.169; FL_c: 8.370; Grad: 5.3
  8%|███▍                                     | 87/1034 [04:14<46:11,  2.93s/it]G: -1.0; G_z: 1.2; G_vp: 9.3; D: -0.6; D_z: 1.2; D_vp: 8.7; CosD: -0.171; FL_c: 8.219; Grad: 5.6
  9%|███▍                                     | 88/1034 [04:17<46:08,  2.93s/it]G: -0.8; G_z: 1.3; G_vp: 9.5; D: -0.7; D_z: 1.2; D_vp: 9.3; CosD: -0.151; FL_c: 8.514; Grad: 5.0 
```

**Comment by karthikkovalam** [comment]
Hi, if you go into your experiment folder (--exp_name SSV) you will find all intermediate results being generated in the folders sample, gen_samples according to tha parameters you have set in --save_interval and --sample_interval.

**Comment by Guocode** [comment]
> Hi, if you go into your experiment folder (--exp_name SSV) you will find all intermediate results being generated in the folders sample, gen_samples according to tha parameters you have set in --save_interval and --sample_interval.

I failed to reproduce your result, after 50,000 iters, the sample log shows random noised images. I didn't change a lot of your code except add `out = torch.nn.AdaptiveAvgPool2d(1)(out)` in VPNet and  my batch size is 32

**Comment by karthikkovalam** [comment]
Could you please check what happens if you dont make any modifications? 
Are you training it with faces? Did you preprocess the 300WLP data?
After 50k iters you should see images like these.
![050000](https://user-images.githubusercontent.com/5602560/80364712-a85ec600-8886-11ea-8db0-10b303e25b38.png)

**Comment by Guocode** [comment]
With no modification except the dataloader, it works, thanks for your reply!

**Comment by Guocode** [comment]
Another question is that I draw the axis as following
```
def draw_axis(img, pyr, tdx=None, tdy=None, size=100):
    pitch = pyr[0] * np.pi / 180
    yaw = -(pyr[1] * np.pi / 180)
    roll = pyr[2] * np.pi / 180

    if tdx != None and tdy != None:
        tdx = tdx
        tdy = tdy
    else:
        height, width = img.shape[:2]
        tdx = width / 2
        tdy = height / 2

    # X-Axis pointing to right. drawn in red
    x1 = size * (cos(yaw) * cos(roll)) + tdx
    y1 = size * (cos(pitch) * sin(roll) + cos(roll) * sin(pitch) * sin(yaw)) + tdy

    # Y-Axis | drawn in green
    x2 = size * (-cos(yaw) * sin(roll)) + tdx
    y2 = size * (cos(pitch) * cos(roll) - sin(pitch) * sin(yaw) * sin(roll)) + tdy

    # Z-Axis (out of the screen) drawn in blue
    x3 = size * (sin(yaw)) + tdx
    y3 = size * (-cos(yaw) * sin(pitch)) + tdy

    cv2.line(img, (int(tdx), int(tdy)), (int(x1), int(y1)), (0, 0, 255), 3)
    cv2.line(img, (int(tdx), int(tdy)), (int(x2), int(y2)), (0, 255, 0), 3)
    cv2.line(img, (int(tdx), int(tdy)), (int(x3), int(y3)), (255, 0, 0), 2)

    return img
```
I use the a,e,t as yaw, pitch ,roll , but it seems not correct.

**Comment by Guocode** [comment]
My generation sample looks ok like this.
![109000](https://user-images.githubusercontent.com/29799967/80663648-28736e80-8ac7-11ea-89b1-138c63ec353b.jpg)
But when I try to evaluate the euler angle it looks bad, I don't know whether it's because of my draw code or the discriminater is not well trained

**Comment by karthikkovalam** [comment]
Hi, there is a demo script for plotting the predicted angles. https://github.com/NVlabs/SSV/blob/master/ssv_demo.py
Please follow this for plotting the predictions.
